src/py21cmfast/src/HaloBox.c

```diff
@@ -784,5 +806,105 @@ int ComputeHaloBox(double redshift, struct UserParams *user_params, struct Cosmo
         return(status);
     }
     LOG_DEBUG("Done.");
-    return 0.;
+    return 0;
+}
+
+//test function for getting halo properties from the wrapper, can use a lot of memory for large catalogs
+int halo_props_exposed(double redshift, struct UserParams *user_params, struct CosmoParams *cosmo_params, struct AstroParams *astro_params,
```

Comment:
I don't love having `exposed` in the name of the function -- I would just call this `get_halo_properties` or similar, and _expose_ it.